### PR TITLE
Open up typed topic stats for public inspection 

### PR DESCRIPTION
**Pull request by johanandren**
References #31035

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/pubsub/TopicImpl.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ private[akka] object TopicImpl {
 
   // internal messages, note that the protobuf serializer for those sent remotely is defined in akka-cluster-typed
   final case class GetTopicStats[T](replyTo: ActorRef[TopicStats]) extends Topic.Command[T]
-  final case class TopicStats(localSubscriberCount: Int, topicInstanceCount: Int)
+  final case class TopicStats(localSubscriberCount: Int, topicInstanceCount: Int) extends Topic.TopicStats
```
Comment (patriknw) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/pubsub/TopicImpl.scala:36]: why don't we move `GetTopicStats` and `TopicStats` classes to the public `Topic`?

Comment (johanandren) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/pubsub/TopicImpl.scala:36]: There's a note a few lines up about keeping actual protocol secret for easier bincomp evolve. But maybe less likely that we'd want to add some fields for these messages?

**Review by octonato — APPROVED** [review]
LGTM

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/pubsub/TopicImpl.scala (hunk 1 above)
Comment (patriknw) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/pubsub/TopicImpl.scala:36]: let's keep the current design

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**